### PR TITLE
test(hc): Remove stable parameter from SiloModeTestDecorator

### PR DESCRIPTION
**Pull request by RyanSkonnord**

### CODE REVIEW

File: src/sentry/testutils/silo.py [hunk 1]
```diff
@@ -234,10 +234,9 @@ def apply(self, decorated_obj: Any) -> Any:
             # _silo_modes is used to mark the class as silo decorated in the above validation
             decorated_obj._silo_modes = self.silo_modes
 
-        # Only run non monolith tests when they are marked stable or we are explicitly running for
-        # that mode.
-        if SENTRY_USE_MONOLITH_DBS or not (self.stable or settings.FORCE_SILOED_TESTS):
```
Comment (mrduncan) [inline src/sentry/testutils/silo.py:239]: I think this setting can get removed from `src/sentry/conf/server.py` as well

**Review by mrduncan — APPROVED** [review]

File: tests/sentry/api/endpoints/test_user_avatar.py [hunk 2]
```diff
@@ -164,7 +162,6 @@ def test_put_upload_saves_to_control_file(self):
         assert isinstance(avatar.get_file(), ControlFile)
         assert ControlFile.objects.filter(id=avatar.control_file_id).exists()
 
-    @pytest.mark.skipif(use_split_dbs(), reason="requires single/monolith db mode")
     @mock.patch("sentry.tasks.files.copy_file_to_control_and_update_model.apply_async")
     def test_get_upload_use_control_during_update(self, mock_task):
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_user_avatar.py:166]: This test could be removed now. We don't need to test the intermediary step anymore.

File: src/sentry/testutils/silo.py [hunk 3]
```diff
@@ -85,22 +83,25 @@ def strip_silo_mode_test_suffix(name: str) -> str:
...
+    If a test is marked with both control and region modes, then the primary run will
+    be in monolith mode and a secondary run will be generated in each silo mode.
+
+    When testing on more than one mode, if the decorator is on a test case class,
     an additional class is dynamically generated and added to the module for Pytest
     to pick up. For example, if you write
 
     ```
-        @control_silo_test
+        @control_silo_test(include_monolith_run=True)
```
Comment (azaslavsky) [inline src/sentry/testutils/silo.py:97]: For the relocation tests, we want to test in two configurations: Silo mode-only for the default `backend test`, and then using `Monolith` mode AND a single monolith DB for the `monolith-dbs-test` runner. The former tests relocation to/from SaaS, the latter tests relocation to/from self-hosted, which will remain monolithic in all ways for the foreseeable future. Using a monolith DB for testing is particularly important, as we want to check that a single atomic import transaction remains possible for self-hosted users.

I see that in this PR you are changing how the decorators work, and removing `SENTRY_USE_MONOLITH_DBS`. Will the above testing regime still work once we do that?

**Review by azaslavsky — DISMISSED** [review]

**Review by markstory — APPROVED** [review]

**Review by dashed — APPROVED** [review]

File: src/sentry/testutils/silo.py (hunk 3 above)
Comment (RyanSkonnord) [reply src/sentry/testutils/silo.py:99]: Per discussion, we will continue to support `make test-monolith-dbs`.

**Pull request merged**